qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1927 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+r"""Use the pulse builder to program pulse programs with an imperative
+assembly-like syntax.
+
+To begin pulse programming we must first initialize our program builder
+context with :func:`build`, after which we can begin
+adding program statements. For example, below we write a simple program that
+:func:`play`\s a pulse:
+
+.. jupyter-execute::
+
+    from qiskit import execute
+    from qiskit import pulse
+
+    d0 = pulse.DriveChannel(0)
+
+    with pulse.build() as pulse_prog:
+        pulse.play(pulse.Constant(100, 1.0), d0)
+
+    pulse_prog.draw()
+
+    # Execute on a real backend.
+    # job = execute(pulse_prog, backend)
+
+The builder initializes a :class:`pulse.Schedule`, ``pulse_prog``
+and then begins to construct the program within the context. The output pulse
+schedule will survive after the context is exited and can be executed like a
+normal Qiskit program ``qiskit.execute(pulse_prog, backend)``.
+
+Pulse programming has a simple imperative style. There is little need to deal
+with classes such as the :class:`~qiskit.circuit.QuantumCircuit` or
+:class:`pulse.Schedule`\s. The pulse builder dynamically constructs these
```

Comment:
If people are coming here for the first time, they might find this confusing. Why start with explaining what they don't need to know?

Reply:
I guess this was more for people transitioning to the new interface? Do you think we should handle this case separately? I imagine that I will have to do a docs only PR to restructure the docs to push people towards this way of programming

Reply:
Yes, I think it should be handled separately. That way these docs will be solid for further into the future

Reply:
Fixed in 71ddec3a0